### DISCUSSION TITLE
[질문] JWT를 기반으로 활용한 인증 방식은 Stateless인가?

### DISCUSSION THREAD

**Discussion by hyeonic** [woowacourse-study/2022-http-web-basic-for-all-developer#20] [Q&A]
HTTP는 기본적으로 Stateless(무상태 프로토콜)이다. 즉 서버에서 클라이언트의 상태를 보관하지 않기 때문에 `로그인`과 같은 기능을 개발하기 어렵다. 로그인한 사용자의 경우 로그인을 진행했다는 상태를 서버에서 유지해야 하기 때문이다. 그렇기 때문에 추가적인 브라우저 쿠키와 서버 세션을 통해 서버 자원을 활용하여 상태를 유지해야 한다. 

단순한 JWT의 경우는 클라이언트가 해당 토큰 정보를 가지고 있다. 서버의 경우 클라이언트가 보내는 토큰 정보를 기반으로 유저에 대한 인증을 진행한다. 결국 서버 측에서 로그인 유지를 위한 상태 정보를 가지지 않아도 인증을 진행할 수 있다. 이것은 Stateless 하다고 봐도 괜찮은가?

**Comment by kth990303** [comment]
Jwt 기반 인증방식은 Stateless 하다고 봐도 괜찮을 것 같아요. Access token이 만료돼도, 유효기간이 긴 Refresh token을 통해 인증을 진행해주네요. HTTP도 stateless하지만 지속 연결이 존재하는 것처럼 jwt도 stateless하다고 보는 의견이 많은 것 같습니다!

참고글: [jwt 토큰과 무상태성 블로그 글](https://velog.io/@yena1025/JWT-%ED%86%A0%ED%81%B0%EA%B3%BC-%EB%AC%B4%EC%83%81%ED%83%9C%EC%84%B1Stateless)

**Reply by hyeonic** [reply]
만약에 refresh token을 도입하게 되면 서버 측에서도 refresh token을 관리해야 하는 것으로 알고 있어요! 그렇다면 이때는 특정한 토큰 상태를 저장하기 때문에 Stateless 하지 못하다고 생각해야 할까요?

**Reply by kth990303** [reply]
refresh token도 유효기간이 정해져있기 때문에 stateless 성질을 가지고 있다고 생각합니다! 그 기간동안 상태를 유지하고 이후에 지워버려서 http 지속 연결과 유사하다고 생각했어요 ㅎㅎ 다른 크루분들 의견도 들어보고 싶네요
